ulid.go

```diff
@@ -238,6 +242,12 @@ func (id *ULID) UnmarshalText(v []byte) error {
 		return ErrDataSize
 	}
 
+	// Overflow checking.
+	// See https://github.com/oklog/ulid/issues/9
+	if v[0] > '7' {
```

Comment:
I don’t have the full details of the encoding in mind. Would you run a small test to check that this condition is equivalent to comparing to the whole max ULID string?

Reply:
I don't know if this is exactly what you had in mind, but [here is an example program](https://gist.github.com/peterbourgon/463abddc89e7fc0639de81fa16be2856) that parses a variety of ULIDs.

Here is the output with package ulid on master.

```
01C8GR9WEZ0000000000000000:  1 62 21 84 f1 df  0  0  0  0  0  0  0  0  0  0  (<nil>)
01C8GR9WF00000000000000000:  1 62 21 84 f1 e0  0  0  0  0  0  0  0  0  0  0  (<nil>)
01C8GR9WF10000000000000000:  1 62 21 84 f1 e1  0  0  0  0  0  0  0  0  0  0  (<nil>)
700000000F000000000000000Y: e0  0  0  0  0  f  0  0  0  0  0  0  0  0  0 1e  (<nil>)
700000000F000000000000000Z: e0  0  0  0  0  f  0  0  0  0  0  0  0  0  0 1f  (<nil>)
700000000F0000000000000010: e0  0  0  0  0  f  0  0  0  0  0  0  0  0  0 20  (<nil>)
7ZZZZZZZZX0000000000000000: ff ff ff ff ff fd  0  0  0  0  0  0  0  0  0  0  (<nil>)
7ZZZZZZZZY0000000000000000: ff ff ff ff ff fe  0  0  0  0  0  0  0  0  0  0  (<nil>)
7ZZZZZZZZZ0000000000000000: ff ff ff ff ff ff  0  0  0  0  0  0  0  0  0  0  (<nil>)
7ZZZZZZZZZ0000000000000001: ff ff ff ff ff ff  0  0  0  0  0  0  0  0  0  1  (<nil>)
7ZZZZZZZZZ0000000000000002: ff ff ff ff ff ff  0  0  0  0  0  0  0  0  0  2  (<nil>)
7ZZZZZZZZZ000000000000000Y: ff ff ff ff ff ff  0  0  0  0  0  0  0  0  0 1e  (<nil>)
7ZZZZZZZZZ000000000000000Z: ff ff ff ff ff ff  0  0  0  0  0  0  0  0  0 1f  (<nil>)
7ZZZZZZZZZ0000000000000010: ff ff ff ff ff ff  0  0  0  0  0  0  0  0  0 20  (<nil>)
7ZZZZZZZZZZZZZZZZZZZZZZZZZ: ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff  (<nil>)
80000000000000000000000000:  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  (<nil>)
80000000000000000000000001:  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  1  (<nil>)
80000000000000000000000002:  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  2  (<nil>)
80000000010000000000000000:  0  0  0  0  0  1  0  0  0  0  0  0  0  0  0  0  (<nil>)
80000000020000000000000000:  0  0  0  0  0  2  0  0  0  0  0  0  0  0  0  0  (<nil>)
800000000Y0000000000000000:  0  0  0  0  0 1e  0  0  0  0  0  0  0  0  0  0  (<nil>)
800000000Z0000000000000000:  0  0  0  0  0 1f  0  0  0  0  0  0  0  0  0  0  (<nil>)
80000000100000000000000000:  0  0  0  0  0 20  0  0  0  0  0  0  0  0  0  0  (<nil>)
90000000000000000000000000: 20  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  (<nil>)
Z0000000000000000000000000: e0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  (<nil>)
ZZ000000000000000000000000: ff  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  (<nil>)
ZZZ00000000000000000000000: ff f8  0  0  0  0  0  0  0  0  0  0  0  0  0  0  (<nil>)
ZZZZZZZZZZZZZZZZZZZZZZZZZZ: ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff  (<nil>)
```

Here is the output with package ulid on this branch, overflow.

```
01C8GR9WEZ0000000000000000:  1 62 21 84 f1 df  0  0  0  0  0  0  0  0  0  0  (<nil>)
01C8GR9WF00000000000000000:  1 62 21 84 f1 e0  0  0  0  0  0  0  0  0  0  0  (<nil>)
01C8GR9WF10000000000000000:  1 62 21 84 f1 e1  0  0  0  0  0  0  0  0  0  0  (<nil>)
700000000F000000000000000Y: e0  0  0  0  0  f  0  0  0  0  0  0  0  0  0 1e  (<nil>)
700000000F000000000000000Z: e0  0  0  0  0  f  0  0  0  0  0  0  0  0  0 1f  (<nil>)
700000000F0000000000000010: e0  0  0  0  0  f  0  0  0  0  0  0  0  0  0 20  (<nil>)
7ZZZZZZZZX0000000000000000: ff ff ff ff ff fd  0  0  0  0  0  0  0  0  0  0  (<nil>)
7ZZZZZZZZY0000000000000000: ff ff ff ff ff fe  0  0  0  0  0  0  0  0  0  0  (<nil>)
7ZZZZZZZZZ0000000000000000: ff ff ff ff ff ff  0  0  0  0  0  0  0  0  0  0  (<nil>)
7ZZZZZZZZZ0000000000000001: ff ff ff ff ff ff  0  0  0  0  0  0  0  0  0  1  (<nil>)
7ZZZZZZZZZ0000000000000002: ff ff ff ff ff ff  0  0  0  0  0  0  0  0  0  2  (<nil>)
7ZZZZZZZZZ000000000000000Y: ff ff ff ff ff ff  0  0  0  0  0  0  0  0  0 1e  (<nil>)
7ZZZZZZZZZ000000000000000Z: ff ff ff ff ff ff  0  0  0  0  0  0  0  0  0 1f  (<nil>)
7ZZZZZZZZZ0000000000000010: ff ff ff ff ff ff  0  0  0  0  0  0  0  0  0 20  (<nil>)
7ZZZZZZZZZZZZZZZZZZZZZZZZZ: ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff  (<nil>)
80000000000000000000000000:  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  (ulid: overflow when unmarshaling)
80000000000000000000000001:  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  (ulid: overflow when unmarshaling)
80000000000000000000000002:  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  (ulid: overflow when unmarshaling)
80000000010000000000000000:  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  (ulid: overflow when unmarshaling)
80000000020000000000000000:  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  (ulid: overflow when unmarshaling)
800000000Y0000000000000000:  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  (ulid: overflow when unmarshaling)
800000000Z0000000000000000:  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  (ulid: overflow when unmarshaling)
80000000100000000000000000:  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  (ulid: overflow when unmarshaling)
90000000000000000000000000:  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  (ulid: overflow when unmarshaling)
Z0000000000000000000000000:  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  (ulid: overflow when unmarshaling)
ZZ000000000000000000000000:  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  (ulid: overflow when unmarshaling)
ZZZ00000000000000000000000:  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  (ulid: overflow when unmarshaling)
ZZZZZZZZZZZZZZZZZZZZZZZZZZ:  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  (ulid: overflow when unmarshaling)
```

Does this answer your question?

Reply:
Looks good